src/main/java/com/achdev/onlinebookstoreapp/service/impl/BookServiceImpl.java

```diff
@@ -36,4 +30,27 @@ public BookDto findById(Long id) {
                 .map(bookMapper::toDto)
                 .orElseThrow(() -> new EntityNotFoundException("Can't find book by id: " + id));
     }
+
+    @Override
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        return bookMapper.toDto(bookRepository.save(book));
+    }
+
+    @Override
+    public BookDto updateById(Long id, CreateBookRequestDto requestDto) {
+        return bookRepository.findById(id)
```

Comment:
Let's simplify this code (stream works slowly) 
```suggestion
        Book book = bookRepository.findById(id)
                .orElseThrow(() -> new EntityNotFoundException("Can't update book by id: " + id)));
        bookMapper.updateBookFromDto(requestDto, book);
        return bookMapper.toDto(bookRepository.save(book));
```

Reply:
Ok, thanks